examples/simulation.rs

```diff
@@ -299,13 +299,13 @@ where
             .values()
             .filter_map(TestNode::next_event_time)
             .min()?;
-        let min_ids: Vec<NodeId> = self
+        let next_id = self
             .nodes
             .iter()
             .filter(|(_, node)| node.next_event_time() == Some(min_time))
+            .choose(rng)
```

Comment:
If I am reading the [rand source](https://docs.rs/rand/0.6.1/src/rand/seq/mod.rs.html#173-308) correctly, this will consume `n` random floats. This is only an example, but I assume that usually random numbers are more expensive than the space to store a few node ids and calling choose on the slice.

It reads much nicer though, so I am conflicted! Still worth it though to find out about the *P = 1/n* trick for successively selecting a random element.

Reply:
Good point! Since it's only the simulation example I'm not too worried about the performance or entropy here, so I'd go with the nicer code?

Reply:
The issue with examples is that they, at some point, end up as production code, by virtue of someone being unfamiliar with the specifics using them as, well, an example worth emulating.

At worst, this is `n` additional syscalls. I am somewhat thinking that maybe the `.choose` method on iterators is intended for cases where iterator result are costly to store only.

I'll leave this up to you, I am fine with either.